fakeldap.py

```diff
@@ -417,12 +417,21 @@ def _search_s(self, base, scope, filterstr, attrlist, attrsonly):
             return [(base, attrs)]
         elif scope == self.SCOPE_ONELEVEL:
             simple_query_regex = r"\(\w+=.+\)$"  # matches things like (some_attribute=value)
-            r = re.compile(simple_query_regex)
-            if r.match(filterstr) is None:  # only this very simple search is supported
+            r_simple = re.compile(simple_query_regex)
+
+            anding_query_regex = r"\(\&(\(\w+=.+\))+\)$"  # matches things like (&(some_attribute=value)(other_attr=othervalue)), but with any number of "anded" conditions
+            r_anded = re.compile(anding_query_regex)
+
+            if r_simple.match(filterstr) is not None:
+                search_attr_name, search_attr_value = filterstr[1:-1].split('=')
+                return self._multiple_attrs_onelevel_search(base, [(search_attr_name, search_attr_value)])
+            elif r_anded.match(filterstr) is not None:
+                bracketList = filterstr[3:-2] # cut of the `(&(` and `))` --> attr1=val1)(attr2=val2
+                attrs_conditions = [tuple(condition.split("=")) for condition in bracketList.split(")(")]
+                return self._multiple_attrs_onelevel_search(base, attrs_conditions)
```

Comment:
This block would be much better written with Python named capture groups: https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups

Reply:
It is not trivial to capture multiple groups, so I would rather keep the current version (I will force push a quick fix though since the expressions is a bit buggy).

the `regex` module does have a `.captures()` method to capture multiple occurrences, but the code currently uses re.

Reference:
https://stackoverflow.com/questions/5060659/regexes-how-to-access-multiple-matches-of-a-group
https://stackoverflow.com/questions/4963691/regex-with-multiple-groups

Reply:
OK, probably not worth the effort to rework.